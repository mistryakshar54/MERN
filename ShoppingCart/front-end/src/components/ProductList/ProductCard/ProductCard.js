import React from 'react';
import Card from "react-bootstrap/Card";
import Button from 'react-bootstrap/Button';
import "./ProductCard.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faShoppingCart, faStar } from "@fortawesome/free-solid-svg-icons";

const ProductCardComponent = ( props ) => {
  const onAddToCartHandler = (productData) =>{
    props.onAddToCart(productData);
  }
    return (
      <Card className="product-card" style={{ width: "20rem" }}>
        <Card.Img variant="top" src={props.product.image} />
        <Card.Body>
          <div className="col-12 product-label">
            <label className="col-7 product-name text-left">
              {props.product.name}
            </label>
            <label className="col-5 product-price text-right">
              {props.product.price}&nbsp;{props.product.currency}
              {/* <span className="text-muted">kg</span> */}
            </label>
          </div>
          <div className="col-12">
            <label className="col-6 product-rating text-left">
              <FontAwesomeIcon icon={faStar} />
              {props.product.rating}
            </label>
            <Button
              onClick={() => onAddToCartHandler(props.product)}
              variant="danger"
              id="addToCartBtn"
              className="col-6 add-to-cart"
            >
              <FontAwesomeIcon icon={faShoppingCart} />
              &nbsp; Add to Cart
            </Button>
          </div>
        </Card.Body>
      </Card>
    );
}

export default ProductCardComponent