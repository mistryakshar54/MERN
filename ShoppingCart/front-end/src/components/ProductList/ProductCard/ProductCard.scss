.card
{
    margin: 2% 2%;
    border-radius: 0px;
}
.card-img-top {
    width: auto!important;
    max-height: 300px;
}
.product-card:hover
{
    cursor: pointer;
    transition: 0.3s ease-in-out;
    box-shadow: 1px 1px 12px 0px #d2d2d2;
}
.product-name{
    font-weight: bold;
    font-size: 1.7em;
    margin-bottom: 0px
}
.product-price
{
    color: #dc3545;
    font-size: 1.2em;
    font-weight: 500;
    margin-bottom: 5px;
}