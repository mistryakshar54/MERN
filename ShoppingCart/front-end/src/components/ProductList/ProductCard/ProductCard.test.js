import React from "react";
import { shallow,mount } from "enzyme";
import ProductCard from "./ProductCard";
import "../../../setupTests";
let product = {
               id: "1237",
               name: "Recliner",
               price: 50000,
               currency: "Rs",
               description: "Fresh Pear",
               rating: 3.5,
               image: "images/recliner.jpg"
             };
const cardComponent = mount(<ProductCard product={product} />);

it("Renders product card component correctly " , ()=>{
    expect(cardComponent).toMatchSnapshot();
})
it("Renders the Product Name Correctly ", () => {
  expect(cardComponent.find(".product-name").text()).toEqual("Recliner");
  expect(cardComponent).toMatchSnapshot();
});