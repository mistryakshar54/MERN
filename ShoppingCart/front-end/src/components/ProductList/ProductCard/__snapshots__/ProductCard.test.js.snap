// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders product card component correctly  1`] = `ReactWrapper {}`;

exports[`Renders the Product Name Correctly  1`] = `ReactWrapper {}`;
